velox/type/StringView.h

```diff
@@ -55,8 +55,8 @@ struct StringView {
     memset(this, 0, sizeof(StringView));
   }
 
-  StringView(const char* data, size_t len) : size_(len) {
-    VELOX_DCHECK(data || size_ == 0);
+  StringView(const char* data, int32_t len) : size_(len) {
+    VELOX_CHECK((size_ >= 0 && data) || (!data && size_ == 0));
```

Comment:
I'm concerned that this might make this constructor large enough that it won't get inlined anymore. Do we have a StringView benchmark suite to verify this won't slowdown constructions? If we don't have one yet, having a benchmark suite for StringView would probably be useful to the work on alignment it with Arrow.

Reply:
would something like this make the intention a bit clearer?
```
VELOX_CHECK_GE(size_, 0);
VELOX_DCHECK(data || size_ == 0);
```

Reply:
The dcheck means the test that runs EXPECT_THROW(StringView(nullptr, 10), VeloxException); wont throw in release , guess I should remove that test then..